clang/unittests/Format/FormatTest.cpp

```diff
@@ -28104,6 +28104,251 @@ TEST_F(FormatTest, BreakBinaryOperations) {
                Style);
 }
 
+TEST_F(FormatTest, WrapNamespaceBodyWithEmptyLinesNever) {
+  FormatStyle Style = getLLVMStyle();
+  Style.FixNamespaceComments = false;
+  Style.ShortNamespaceLines = 0;
+  Style.MaxEmptyLinesToKeep = 2;
+  Style.WrapNamespaceBodyWithEmptyLines = FormatStyle::WNBWELS_Never;
+  Style.CompactNamespaces = false;
+
+  // Empty namespace
+  verifyNoChange("namespace N {};", Style);
+
+  // Single namespace
+  verifyNoChange("namespace N {\n"
+                 "int f1(int a) { return 2 * a; }\n"
+                 "};",
+                 Style);
+
+  // Nested namespace
+  verifyNoChange("namespace N1 {\n"
```

Comment:
why verifyNoChange?

Reply:
Used `verifyNoChange` because `verifyFormat` has a strange behaviour with newlines and for no reasons tests were failing with it even though compiled `clang-format` with the same config formatted code exactly as I expected it. Also my option is quite interconnected with `MaxEmptyLinesToKeep`, whose behaviour is mostly tested with `verifyNoChange` and it works as expected

Reply:
Ok, but the idea is that it messes with code as if the person was editing a file and this was the first clang-format, I've never needed to use `verifyNoChange` tests failing often meant I wasn't doing the correct thing

Reply:
But are there any differences between using `verifyNoChange` and `verifyFormat` with one string argument?

Reply:
For `MaxEmptyLinesToKeep` we can't use `messUp`, since we only want to _keep_ the empty lines, we never want to add more than one.